src/components/RenderRouter/BlogItem.tsx

```diff
@@ -0,0 +1,94 @@
+
+import React from 'react';
+import { withRouter, RouteComponentProps } from 'react-router-dom';
+import { Button } from 'reactstrap';
+import PropTypes from "prop-types";
+import { withCookies } from 'react-cookie';
+import "./BlogItem.scss"
+import * as queries from '../../graphql/queries';
+import Amplify, { API } from 'aws-amplify';
+import awsmobile from '../../aws-exports';
+import format from 'date-fns/format';
+import { GRAPHQL_AUTH_MODE } from '@aws-amplify/api/lib/types';
+
+//needs an changes from video content to blog content...
+
+//import uuidv4 from 'uuid/v4'
+Amplify.configure(awsmobile);
+
+interface Props extends RouteComponentProps {
+    content: any,
+    cookies: any
+}
+interface State {
+    content: any,
+    listData: any,
+    overlayData: any,
+    publishedOnly: any
+}
+class BlogItem extends React.Component<Props, State> {
+
+    static contextTypes = {
+        router: PropTypes.object,
+        history: PropTypes.object
+    }
+
+    constructor(props: Props, context: any) {
+        super(props, context);
+        console.log(context);
+        this.state = {
+            content: props.content,
+            listData: null,
+            overlayData: null,
+            publishedOnly: null
+        }
+        const getBlogByBlogStatus:any = API.graphql({
+            query: queries.getBlogByBlogStatus,
+            variables: { blogStatus: this.state.content.status, sortDirection: this.state.content.sortOrder },
+            authMode: GRAPHQL_AUTH_MODE.API_KEY
+        });
+        getBlogByBlogStatus.then((json: any) => {
+            console.log("Success queries.getBlogByBlogStatus: " + json);
+            console.log(json)
+            this.setState({
+                listData: json.data.getBlogByBlogStatus.items
+            })
+            const today = format(new Date(), "yyyy-MM-dd")
+            var dateChecked = this.state.listData.filter((post: any) => post.publishedDate <= today && (post.expirationDate >= today || post.expirationDate))
```

Comment:
@GeorgeBellTMH this is something I'm not quite happy with. Since DynamoDB only lets you query by two fields at a time, I can't query for posts that are live, and are within the publishedDate and expirationDate constraints, so I have to filter. Could we setup a Lambda function to set posts as live on the publishedDate and unlisted after the expirationDate?